PolyclinicLab/Polyclinic.Application/Services/AnalyticsService.cs

```diff
@@ -0,0 +1,116 @@
+﻿using AutoMapper;
+using Polyclinic.Application.Contracts;
+using Polyclinic.Application.Interfaces;
+using Polyclinic.Domain.Models;
+using Polyclinic.Domain.Interfaces;
+
+namespace Polyclinic.Application.Services;
+
+public class AnalyticsService(
+    IRepository<Doctor, int> doctorRepo,
+    IRepository<Appointment, int> appointmentRepo,
+    IMapper mapper
+) : IAnalyticsService
+{
+    // (1) Doctors with >=10 years of experience
+    public List<DoctorDto> GetExperiencedDoctors(int minExperience)
+    {
+        try
+        {
+            var allDoctors = doctorRepo.ReadAll();
+            Console.WriteLine($"[DEBUG] Total doctors in database: {allDoctors.Count}");
+            
+            var doctors = allDoctors
+                .Where(d => d.Experience >= minExperience)
+                .ToList();
+            
+            Console.WriteLine($"[DEBUG] Doctors with experience >= {minExperience}: {doctors.Count}");
+            return mapper.Map<List<DoctorDto>>(doctors);
+        }
+        catch (Exception ex)
+        {
+            Console.WriteLine($"[ERROR] GetExperiencedDoctors failed: {ex.Message}");
+            Console.WriteLine($"[ERROR] Stack trace: {ex.StackTrace}");
+            throw;
+        }
+    }
+
+    public bool DoctorExists(int doctorId)
+    {
+        return doctorRepo.ReadAll().Any(d => d.Id == doctorId);
+    }
+
+    // (2) Patients who visited a specific doctor, ordered by name
+    public List<PatientDto> GetPatientsByDoctor(int doctorId)
+    {
+        var patients = appointmentRepo.ReadAll()
+            .Where(a => a.Doctor.Id == doctorId)
```

Comment:
Doctor может быть null

```suggestion
            .Where(a => a.Doctor?.Id == doctorId)
```